presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -643,6 +655,37 @@ public void dropColumn(String databaseName, String tableName, String columnName)
     }
 
     private List<Partition> getPartitions(String databaseName, String tableName, String expression)
+    {
+        if (numSegments == 1) {
+            return getPartitions(databaseName, tableName, expression, null);
+        }
+
+        // Do parallel partition fetch.
+        CompletionService<List<Partition>> completionService = new ExecutorCompletionService<>(executor);
+        List<Segment> segments = IntStream.range(0, numSegments)
+                .mapToObj(s -> new Segment().withSegmentNumber(s).withTotalSegments(numSegments))
+                .collect(toList());
```

Comment:
we prefer immutable collections, so: `toImmutableList()` 

however, in this case you don't need the list at all:
```
.forEach(segment -> completionService.submit(() -> ....);
```

Reply:
OK